core/controllers/access_validators_test.py

```diff
@@ -779,3 +781,43 @@ def test_should_not_access_for_logged_out_user(self) -> None:
             ACCESS_VALIDATION_HANDLER_PREFIX,
             ), expected_status_int=404
         )
+
+
+class StoryEditorPageAccessValidationHandlerTests(test_utils.GenericTestBase):
+    """Checks the access to the story editor page and its rendering."""
+
+    def setUp(self) -> None:
+        super().setUp()
+        self.signup(self.CURRICULUM_ADMIN_EMAIL, self.CURRICULUM_ADMIN_USERNAME)
+        self.add_user_role(
+            self.CURRICULUM_ADMIN_USERNAME, feconf.ROLE_ID_CURRICULUM_ADMIN)
+
+        self.admin_id = self.get_user_id_from_email(self.CURRICULUM_ADMIN_EMAIL)
+
+        self.topic_id = topic_fetchers.get_new_topic_id()
+        self.story_id = story_services.get_new_story_id()
+        self.save_new_story(
+            self.story_id, self.admin_id, self.topic_id)
+        self.save_new_topic(
+            self.topic_id, self.admin_id, name='Name',
+            abbreviated_name='topic-one', url_fragment='topic-one',
+            description='Description', canonical_story_ids=[self.story_id],
+            additional_story_ids=[],
+            uncategorized_skill_ids=[],
+            subtopics=[], next_subtopic_id=1)
+
+    def test_access_story_editor_page_without_logging_in(self) -> None:
+        self.get_json(
+            '%s/can_access_story_editor_page/%s' % (
+                ACCESS_VALIDATION_HANDLER_PREFIX, self.topic_id
+            ), expected_status_int=401
+        )
+
+    def test_access_story_editor_page_with_curriculum_admin(
+            self) -> None:
+        self.login(self.CURRICULUM_ADMIN_EMAIL)
+        self.get_html_response(
```

Comment:
Why get_html_response and not get_json?

Reply:
Hey @DubeySandeep, the last two merged PRs for this issue both used get_html_response (#20295 , #20116). To maintain uniformity across all PRs under this issue, I used the same. Should I change it to get_json?

Reply:
Using get_html... seems unexpected here, can you please use get_json?

Reply:
Done